### PR TITLE
Add sig-cluster-lifecycle@kubernetes.io group, update emails/lists

### PR DESCRIPTION
**Pull request by vincepri**

### CODE REVIEW

File: groups/sig-cluster-lifecycle/groups.yaml [hunk 1]
```diff
@@ -7,6 +7,28 @@ groups:
   # and is not intended to govern access to infrastructure
   #
 
+  - email-id: sig-cluster-lifecycle@kubernetes.io
+    name: sig-cluster-lifecycle
+    description: |
+      Group dedicated to improving the Kubernetes contributor experience.
```
Comment (ameukam) [inline groups/sig-cluster-lifecycle/groups.yaml:13]: Copy/Paste ?

Comment (vincepri) [reply groups/sig-cluster-lifecycle/groups.yaml:13]: Yep, fixing :D

**Comment by fabriziopandini** [comment]
+1

**Review by neolit123 — COMMENTED** [review]
/lgtm
/approve
+1 to migrating the ML

**Pull request merged**